src/apiService/customClass.js

```diff
@@ -14,33 +14,40 @@ export class DeviceInfo {
     team,
   }) {
     this.createdTime = new Date();
-    this.computerCompanyName = computerCompanyName;
-    this.computerConfig = computerConfig;
-    this.computersSeriNumber = computersSeriNumber;
-    this.mouseCompanyName = mouseCompanyName;
     this.name = name;
-    this.numberOfMouse = numberOfMouse || 0;
-    this.numberOfScreen = numberOfScreen || 0;
-    this.screenConfig = screenConfig || 0;
-    this.screenSize = screenSize;
     this.team = team;
+    this.computer = {
+      companyName: computerCompanyName,
+      config: computerConfig,
+      seriNumber: computersSeriNumber,
+    };
+    this.mouse = {
+      companyName: mouseCompanyName,
+      numberOf: numberOfMouse || 0,
+    };
+    this.screen = {
+      numberOf: numberOfScreen || 0,
+      config: screenConfig || 0,
+      size: screenSize,
+    };
   }
 }
 
 export const deviceInfoConverter = {
   toFirestore: (deviceInfo) => {
     return {
       name: deviceInfo.name,
-      computerCompanyName: deviceInfo.computerCompanyName,
-      computerConfig: deviceInfo.computerConfig,
-      computersSeriNumber: deviceInfo.computersSeriNumber,
-      mouseCompanyName: deviceInfo.mouseCompanyName,
-      numberOfMouse: deviceInfo.numberOfMouse,
-      numberOfScreen: deviceInfo.numberOfScreen,
-      screenConfig: deviceInfo.screenConfig,
-      screenSize: deviceInfo.screenSize,
       team: deviceInfo.team,
       createdTime: deviceInfo.createdTime,
+      computer: {
+        ...deviceInfo.computer,
+      },
+      mouse: {
+        ...deviceInfo.mouse,
+      },
+      screen: {
+        ...deviceInfo.screen,
+      },
```

Comment:
What is the purpose of using spread syntax?

Reply:
It makes copies for all properties of an object to another. So you don't need to write each property and assign it

Reply:
From that point, can we just return `deviceInfo` instead?

Reply:
In my opinion, I want to make a deep copy of deviceInfo which doesn't have any relative to the old device info. If return deviceInfo, computer, mouse, screen properties will become a shallow copy and have reference to old deviceInfo

Reply:
So how about using spread syntax on the `deviceInfo`? Such as ` return { ...deviceInfo }

Reply:
spread operator deep copies the data if it is not nested and `deviceInfo, computer, mouse, screen` is nested. We can use it with deviceRequest` below